strawberry_django_plus/middlewares/user_warmup.py

```diff
@@ -0,0 +1,27 @@
+import inspect
+from typing import Any, Awaitable, Callable, Coroutine, Union, cast
+
+from asgiref.sync import sync_to_async
+from django.http import HttpRequest, HttpResponse
+from django.utils.decorators import sync_and_async_middleware
+
+
+@sync_and_async_middleware
+def user_warmup_middleware(
+    get_response: Callable[[HttpRequest], Union[HttpResponse, Coroutine[Any, Any, HttpResponse]]],
+):
+    if inspect.iscoroutinefunction(get_response):
+
+        async def middleware(request):  # type: ignore
+            # Warm up user object in sync context
+            await sync_to_async(getattr)(request.user, "is_anonymous")
```

Comment:
Oh, I let that pass in the latest review

If you do `sync_to_async(getattr)(request.user, "is_anonymous")`, you are still retrieving the user outside of the async safe context. You should instead do this:

```suggestion
            await sync_to_async(getattr)(request, "user")
```

Also no need to access `is_anonymous` in this case (unless you want to). Then would be:

```suggestion
            (await sync_to_async(getattr)(request, "user")).is_anonymous
```

You will probably need to add a `noqa` in here as well

Reply:
Retrieving the user in the async context is not a problem as it only gets the LazyUserObject. The sync methods are only called once you access any methods, such as `is_anonymous`.  You can check it with the `test_query_connection_get_queryset[async]` test

Reply:
Oh, you are correct! For some reason I thought that just retrieving the `user` from the request would trigger a database call, but it actually only happens when accessing anything from it. TIL :)